packages/gatsby-plugin-gatsby-cloud/src/components/Indicator.js

```diff
@@ -85,19 +163,92 @@ export default function Indicator({ children }) {
     }
   }, [])
 
+  if (buildInfo?.currentBuild?.buildStatus === `BUILDING`) {
+    return (
+      <PreviewIndicator>
+        <GatsbyIndicatorButton
+          tooltipText={`Building a new preview`}
+          showSpinner={true}
+          overrideShowTooltip={true}
+        />
+        <LinkIndicatorButton tooltipText={`Copy link`} active={true} />
+        <InfoIndicatorButton />
+      </PreviewIndicator>
+    )
+  }
+
+  if (buildInfo?.currentBuild?.buildStatus === `ERROR`) {
+    return (
+      <PreviewIndicator>
+        <GatsbyIndicatorButton
+          tooltipText={`Preview error`}
+          overrideShowTooltip={true}
+          active={true}
+          tooltipIcon={failedIcon}
+          tooltipLink={`View logs`}
+          tooltipLinkImage={logsIcon}
+          onClick={() =>
+            viewLogsClick({
+              orgId: buildInfo?.siteInfo?.orgId,
+              siteId: buildInfo?.siteInfo?.siteId,
+              errorBuildId: buildInfo?.currentBuild?.id,
+            })
+          }
+        />
+        <LinkIndicatorButton />
+        <InfoIndicatorButton />
+      </PreviewIndicator>
+    )
+  }
+
+  if (buildId && buildId === buildInfo?.currentBuild?.id) {
+    const updatedDate = formatDistance(
+      Date.now(),
+      new Date(buildInfo?.currentBuild?.createdAt),
+      { includeSeconds: true }
+    )
+    return (
+      <PreviewIndicator>
+        <GatsbyIndicatorButton active={true} />
+        <LinkIndicatorButton tooltipText={`Copy link`} active={true} />
+        <InfoIndicatorButton
+          tooltipText={`Preview updated ${updatedDate} ago`}
+          active={true}
+        />
+      </PreviewIndicator>
+    )
+  }
+
+  if (
+    buildId &&
+    buildId !== buildInfo?.latestBuild?.id &&
+    buildInfo?.latestBuild?.buildStatus === `SUCCESS`
+  ) {
+    return (
+      <PreviewIndicator>
+        <GatsbyIndicatorButton
+          tooltipText={`New preview available`}
+          overrideShowTooltip={true}
+          active={true}
+          onClick={() =>
+            newPreviewAvailableClick({
+              isOnPrettyUrl: buildInfo?.isOnPrettyUrl,
+              sitePrefix: buildInfo?.siteInfo?.sitePrefix,
+            })
+          }
+          tooltipLink={`Click to view`}
+        />
+        <LinkIndicatorButton />
+        <InfoIndicatorButton />
+      </PreviewIndicator>
+    )
+  }
+
   return (
-    <>
-      <Style />
-      <div
-        data-testid="preview-status-indicator"
-        data-gatsby-preview-indicator="root"
-        aria-live="assertive"
-      >
-        <GatsbyIndicatorButton {...buildInfo} />
-        <LinkIndicatorButton {...buildInfo} />
-        <InfoIndicatorButton {...buildInfo} />
-      </div>
-      {children}
```

Comment:
The new implementation expects children to be instances of `<IndicatorButton />`s

@DanielSLew Is this change safe to make or do we need to account for existing usage of `children`?

Reply:
I believe `children` was just leftover from before I rendered this with a React portal. I don't believe anything relies on this now.